src/app/app.component.ts

```diff
@@ -1,21 +1,25 @@
-import { Component } from '@angular/core';
+import { Component, ViewChild } from '@angular/core';
 
 import { LoggerService } from './logger.service';
+import { HelloComponent } from './hello.component';
 
 @Component({
   selector: 'my-app',
   templateUrl: './app.component.html',
   styleUrls: ['./app.component.css'],
 })
 export class AppComponent {
+  @ViewChild(HelloComponent, {static: true}) childComp: HelloComponent;
   name = 'Angular';
 
   stateLog: string[] = [];
 
   constructor(private loggerService: LoggerService) {}
 
-  onStateChanged(state: boolean) {
-    this.stateLog.push(`State became ${state}`);
-    this.loggerService.log(`AppComponent: New state: ${state}`);
+  ngAfterContentInit(): void {
+    this.childComp.stateChange.subscribe((state: boolean) => {
```

Comment:
This subscription will leak when component is destroyed.

When you perform manual subscriptions to RXJS streams you should always take care to unsubscribe later on component destruction hook `OnDestroy`